crates/services/sync/src/import/tests.rs

```diff
@@ -227,6 +247,204 @@ async fn import__signature_fails_on_header_5_only() {
     assert_eq!((State::new(4, None), false), res);
 }
 
+#[tokio::test]
+async fn import__keep_data_asked_in_fail_ask_header_cases() {
+    let params = Config {
+        block_stream_buffer_size: 10,
+        header_batch_size: 1,
+    };
+
+    let mut consensus_port = MockConsensusPort::default();
+    // No reask
+    consensus_port
+        .expect_check_sealed_header()
+        .times(3)
+        .returning(|_| Ok(true));
+    // No reask
+    consensus_port
+        .expect_await_da_height()
+        .times(3)
+        .returning(|_| Ok(()));
+
+    let mut p2p = MockPeerToPeerPort::default();
+    let mut seq = Sequence::new();
+    // Given
+    // Fail to get headers for block 4
+    p2p.expect_get_sealed_block_headers()
+        .times(1)
+        .in_sequence(&mut seq)
+        .returning(|_| {
+            Box::pin(async move {
+                tokio::time::sleep(Duration::from_millis(300)).await;
+                Err(anyhow::anyhow!("Some network error"))
+            })
+        });
+    p2p.expect_get_sealed_block_headers()
+        .times(2)
+        .in_sequence(&mut seq)
+        .returning(|range| {
+            Box::pin(async move {
+                let peer = random_peer();
+                let headers = Some(range.map(empty_header).collect());
+                let headers = peer.bind(headers);
+                Ok(headers)
+            })
+        });
+    // Then
+    // Reask only for block 4
+    p2p.expect_get_sealed_block_headers()
+        .times(1)
+        .in_sequence(&mut seq)
+        .returning(|range| {
+            Box::pin(async move {
+                let peer = random_peer();
+                let headers = Some(range.map(empty_header).collect());
+                let headers = peer.bind(headers);
+                Ok(headers)
+            })
+        });
```

Comment:
Why the sequence is [fail, success, success]? Based on the comments I will expect either [fail, success] or [success(for first 3 blocks), fail, success]. The same question for get transactions.

Reply:
I have explicit the comments the expected is indeed : [fail (4), success(5), success(6)] and then : [success(4)] only for both of the tests. Tell me if it's more clear :)

Reply:
It would be nice if the closure inside of the `get_sealed_block_headers` verified the block height like `height == 3`. In this case it will be clear which block is `success` and which is `fail`